src/pages/Software.pug

```diff
@@ -2,5 +2,91 @@ extends ../templates/contents.pug
+
+        # Overview
+
+        Our software is written in Python.
+        Right now, it consists of a program library and a command line tool that utilizes the library.
+        The command line tool makes all the basic functionalities of our library available, namely encoding and decoding information.
+        In addition, it would be possible to implement a graphical user interface, which would make the usage even easier for everyone.
+
+        In the following, the two functions of our software encoding and decoding are explained in detail.
+        It will become clear how we are able to store information on DNA, although our synthesis approach is semi-specific.
+        We will also discuss the challenges that arise when decoding information.
+
+        # Encoding Information on DNA
+
+    - var imgpath = "" + require("../assets/img/Software/encoding.svg")["default"]
+    +image(1, imgpath, "Overview over the steps needed for encoding data into DNA.")
+
+    :markdown-it(html)
+
+        ## Changes of Nucleotides as Information Carriers
+
+        Our DNA synthesis is semi-specific, meaning that we cannot control how many nucleotides are attached in each step.
+        This means that we will always have chunks of unknown length of each nucleotide, e.g. "AAACCCCCAAAAAGGGG".
+        These chunks are also called homonucleotides.
+        To still be able to encode information in this, we are using a rather unusual encoding scheme.
+        Instead of encoding one bit of information to each base pair, we are encoding information in each change of bases.
+        For example, the change from a homonucleotide of As to a homonucleotide of Cs could encode a `1`.
+
+        ## Using a Ternary Code
+
+        From each homonucleotide, three other homonucleotides are reachable.
+        From "A" we can go to "C", "G", and "T".
+        From "C" we can go to "A", "G", and "T", and so on.
+        This means that to make use of all possible combinations, we do not need a binary code, but a ternary code.
+        Computers usually work with a binary code, meaning that every information is encoded in bits.
+        Each bit can either be `0` or `1`.
+        Instead, in our ternary code information is encoded in trits.
+        Each trit can have the states `0`, `1`, or `2`.
+
+        ## Converting form Binary to Ternary
+
+        Our software is able to encode arbitrary files into DNA strands.
+        Therefore, the first step is to convert binary information into ternary information.
+        Computers work with bytes, so chunks of eight bits.
+        Hence, it is natural to first split the string of bits we get from a file into bytes.
+        Then, the software converts each byte into trits.
+        Each byte can have $2^8=256$ states.
```

Comment:
```suggestion
        
        The number of values a number can take can be calculated by the formula $b^l$.
        In this $b$ is the base of the number system, e.g., '2' for the binary system, and$ l$ is the number of digits.
        Since a byte consists of eight bits with two states each, every byte can have $2^8=256$ states.
```

Reply:
Ich würde den ersten Satz vllt so schreiben: "The count of values a number can take is calculated by the formula .." Ich weiß nicht, ob man count sagen kann, aber grammarly hat nicht gemeckert und DeepL übersetzt es auch mit Anzahl.
Im zweiten Satz fehlt das Hochstellungszeichen vor der 1
Ansonsten deutlich verständlicher :)